### PR TITLE
Add traffic logging

### LINKED ISSUE #31: Enable users to opt-into a debug/logging mode
We have had several discussions (internally and externally) how to best debug an implementation of enigma.js and we always end up monkey-patching the enigma.js source code to log out e.g. the traffic.

Let's design and implement a good way of allowing users to toggle on debugging/logging.

### PR DESCRIPTION
**Pull request by gabbaxx**
Fixes #31.

### Status

* [ ] Under development
* [x] Waiting for code review
* [ ] Waiting for merge

### Information

* [ ] Contains breaking changes
* [x] Contains new API(s)
* [x] Contains documentation
* [x] Contains test

### To-do list

* [ ] [Fix this thing]
* [ ] [Fix another thing]
* [ ] ...

### CODE REVIEW

**Comment by peol** [comment]
Nice one! Would you mind also adding an example in the configuration documentation? I think this warrants an example of how one can handle logs.

https://github.com/qlik-oss/enigma.js/blob/master/docs/qix/configuration.md
and
https://github.com/qlik-oss/enigma.js/tree/master/docs/rest

File: test/integration/rest-logging.spec.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+describe('rest-logging', () => {
+  let sandbox;
+  let aboutService;
+  let logSpy;
+  before(() => {
+    sandbox = sinon.sandbox.create();
+    logSpy = sandbox.spy();
+
+    const cfg = {
+      unsecure: true,
```
Comment (FredrikFolkesson) [inline test/integration/rest-logging.spec.js]: Change to `secure: false`

File: src/services/rest/http-client.js [hunk 2]
```diff
@@ -132,13 +135,19 @@ export default class HttpClient {
...
     const req = this.httpModule.request(settings);
 
-    if (typeof opts.body !== 'undefined') {
+    const hasBody = typeof opts.body !== 'undefined';
+    if (hasBody) {
       req.write(opts.body);
     }
 
+    if (restOpts.handleLog) {
+      restOpts.handleLog(hasBody ?
```
Comment (peol) [inline src/services/rest/http-client.js]: Maybe `{ msg: 'Sent', data: { request: settings, body: opts.body || null } }`? Or is it important to not expose a `body` property if it doesn't exist?

**Comment by peol** [comment]
LGTM

**Pull request merged**